pkg/registry/client.go

```diff
@@ -0,0 +1,146 @@
+package registry
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"path"
+	"sync"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	orasregistry "oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// Context describes aspects of a registry.
+type Context struct {
```

Comment:
I personally find a bit confusing to name this context given how common `context.Context` is in Go and that this one doesn't implement `context.Context`.

What do you think about about `RegistryConfig`?

Reply:
sounds good to me.

Reply:
lint doesn't like RegistryContext, so I'll go with something else like maybe StorageContext